### PR TITLE
[GDBJIT] Specify C# language in DWARF debug information

### PR DESCRIPTION
**Pull request by kbaladurin**
To correctly support C# language in lldb DWARF debug information
should contain correct information about language.

Now we use 0x9e57 constant for specify C# language.

### CODE REVIEW

**Comment by kbaladurin** [comment]
cc @ayuckhulk @Dmitri-Botcharnikov

**Comment by Dmitri-Botcharnikov** [comment]
cc @chunseoklee @seanshpark

**Comment by kbaladurin** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by kbaladurin** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by janvorli** [comment]
@swgillespie the GC/API/NoGCRegion/NoGC test has failed on Ubuntu x64 here, could you please take a look? Just in case someone re-run the test, here is the link to the console output:
https://ci.dot.net/job/dotnet_coreclr/job/master/job/checked_ubuntu_tst_prtest/9039/consoleFull#2096352037c7af6a29-b465-404a-b249-90a3911e4354

**Comment by swgillespie** [comment]
@janvorli Yep, taking a look today (https://github.com/dotnet/coreclr/issues/13736)

**Comment by kbaladurin** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by janvorli** [comment]
@kbaladurin can you please explain the consequences of this change? Does it have some immediate effect?

**Comment by kbaladurin** [comment]
@janvorli this change allows to separate c# and c++ type formatters in our lldb version that improves mixed  code (c++ and c#) debugging. But vanilla lldb will not able to show local variables and arguments with this patch because it doesn't know anything about c#.

**Comment by janvorli** [comment]
@kbaladurin I am sorry for the delay, I got distracted by debugging some issues. I am a bit hesitant to make this change unconditionally to break lldb ability to show locals and arguments for people that don't have your patched version of lldb. Could you please add a runtime config option to make this optional? Named e.g. GDBJitLangId or something like that. It would contain the language identifier. The default would be the DW_LANG_C89, but if you set GDBJitLangId=9e57, it would override it to your new constant.

**Comment by Dmitri-Botcharnikov** [comment]
@janvorli Could it be configurational feature like FEATURE_GDBJIT_LANGID? We can enable it for Tizen platform and leave as is for the others.

**Comment by janvorli** [comment]
@Dmitri-Botcharnikov that would also work

**Comment by kbaladurin** [comment]
@janvorli thank you for note! I've added  new feature to specify c# identificator on compile time.

File: clrdefinitions.cmake [hunk 1]
```diff
@@ -122,6 +122,9 @@ endif()
 if(FEATURE_GDBJIT_FRAME)
     add_definitions(-DFEATURE_GDBJIT_FRAME)
 endif(FEATURE_GDBJIT_FRAME)
+if(FEATURE_GDBJIT_LANGID)
```
Comment (janvorli) [inline clrdefinitions.cmake]: A nit - all of our FEATURE_XXX definitions (both in CMake and in C++) are value-less. So adding a new one that has a value feels a bit out of line. I would prefer using FEATURE_GDBJIT_LANGID_CS and use it to select whether we embed the DW_LANG_C89 or your new CS langid.

Comment (kbaladurin) [reply clrdefinitions.cmake]: Done. Thank you!

**Comment by kbaladurin** [comment]
@dotnet-bot Tizen armel Cross Debug Build

**Comment by kbaladurin** [comment]
@dotnet-bot test Tizen armel Cross Debug Build

**Comment by kbaladurin** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by kbaladurin** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Pull request merged**